csrc/velox/functions/text/add_tokens.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "velox/functions/Udf.h"
+#include "velox/type/Type.h"
+
+namespace facebook::torcharrow::functions
+{
+    template <typename T>
+    struct add_tokens
+    {
+        VELOX_DEFINE_FUNCTION_TYPES(T);
+
+        template <typename TInput, typename TOutput, typename TTokens>
+        FOLLY_ALWAYS_INLINE void call(
+            TOutput &output,
+            const TInput &input,
+            const TTokens &tokens,
+            bool begin = true)
+        {
+            output.reserve(input.size() + tokens.size());
+            if (begin)
```

Comment:
hmm, that's a good question but I don't think there is such use-case in text AFAIK. But this operator may be consumed beyond text I guess..So if there exist such use-case outside of text, we could potentially add this option :)

Reply:
The benchmark[1] seems to always add two tokens? Perhaps we can have an overloaded version of `add_tokens` that add both beginning and the end to avoid one list allocation? 

[1] https://github.com/pytorch/text/pull/1807/files